src/ralph/discovery/models_device.py

```diff
@@ -877,6 +877,26 @@ def orientation(self):
         from ralph_assets.models import Orientation
         return Orientation.name_from_id(asset.device_info.orientation)
 
+    @property
+    def processors_repr(self):
+        processors_count = self.processor_set.all().count() or 0
+        processors = {dd.model.name for dd in self.processor_set.all()}
+        if len(processors) > 1:
+            raise ValueError("Device has different processors models")
+        elif len(processors) == 0:
+            processors_repr = '-'
+        else:
+            processors_repr = "{processors_count}x {processors_name}".format(
+                processors_count=processors_count,
+                processors_name=list(processors)[0],
+            )
+        return processors_repr
+
+    @property
+    def memory_repr(self):
+        summed = sum([i.size or 0 for i in self.memory_set.all()])
```

Comment:
try ``self.memory_set.all().aggregate(Sum('size'))`` without ``sum()`` and list comprehension